### ISSUE TITLE
Unable to build x86 based unit tests with bazel 6.0.0rc4 because there is no x86 based remote java toolchain

### ISSUE THREAD

**Issue by carpenterjc** [bazelbuild/bazel#16961]
### Description of the bug:


We have a C++ toolchain with both intel x64 and x86 definitions. 
When we build with x64 the tests work fine, when we build with x86 you can see the toolchain resolution doesn't find any remote_jdk_11 with the which matches the CPU. 

```
ERROR: C:/dev/_bazel/jarbmqax/external/remote_coverage_tools/BUILD:10:12: While resolving toolchains for target @remote_coverage_tools//:Main: No matching toolchains found for types @bazel_tools//tools/jdk:runtime_toolchain_type.                                                                                                                                                                                                     
To debug, rerun with --toolchain_resolution_debug='@bazel_tools//tools/jdk:runtime_toolchain_type'                                                                                                                   
If platforms or toolchains are a new concept for you, we'd encourage reading https://bazel.build/concepts/platforms-intro.                                                                                     
```

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.


Unfortunately: You need a C++ toolchain with x64 and x86 support, we are using platforms. 

When but just running bazel build with the x64 platform will work, as soon as we switch the to building with an x86 platform we get the toolchain resolution error. 

We had to specify `build --java_runtime_version=remotejdk_11` to get the build to work.

### Which operating system are you running Bazel on?


Windows

### What is the output of `bazel info release`?


release 6.0.0rc4

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.


_No response_

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

_No response_


### Have you found anything relevant by searching the web?

_No response_

### Any other information, logs, or outputs that you want to share?

```text
bazel version 
`Bazelisk version: v1.15.0
Build label: 6.0.0rc4
Build target: bazel-out/x64_windows-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Fri Dec 2 13:47:04 2022 (1669988824)
Build timestamp: 1669988824
Build timestamp as int: 1669988824
`



INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @local_jdk//:jdk; mismatching config settings: localjdk_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux//:jdk; mismatching values: linux
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_aarch64//:jdk; mismatching values: linux, arm64
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_ppc64le//:jdk; mismatching values: linux, ppc
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_s390x//:jdk; mismatching values: linux, s390x
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_macos//:jdk; mismatching values: osx              
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_macos_aarch64//:jdk; mismatching values: osx, arm64
INFO: ToolchainResolution:   Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: execution @local_config_platform//:host: Selected toolchain @remotejdk11_win//:jdk  
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_win_arm64//:jdk; mismatching values: arm64        
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_linux//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_linux_aarch64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_macos//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_win//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_win_arm64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_linux//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_linux_aarch64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_macos//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_win//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_win_arm64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution: Target platform @local_config_platform//:host: Selected execution platform @local_config_platform//:host, type @bazel_tools//tools/cpp:toolchain_type -> toolchain //tools/config:sophos_win_x64_toolchain, type @bazel_tools//tools/jdk:toolchain_type -> toolchain @bazel_tools//tools/jdk:toolchain_java11, type @bazel_tools//tools/jdk:runtime_toolchain_type -> toolchain @remotejdk11_win//:jdk
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @local_jdk//:jdk; mismatching config settings: localjdk_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux//:jdk; mismatching values: linux            
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_aarch64//:jdk; mismatching values: linux, arm64                                                                                                                                                                                                                  
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_ppc64le//:jdk; mismatching values: linux, ppc
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_s390x//:jdk; mismatching values: linux, s390x                                                                                                                                                                                                                    
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_macos//:jdk; mismatching values: osx              
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_macos_aarch64//:jdk; mismatching values: osx, arm64                                                                                                                                                                                                                    
INFO: ToolchainResolution:   Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: execution @local_config_platform//:host: Selected toolchain @remotejdk11_win//:jdk  
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk11_win_arm64//:jdk; mismatching values: arm64        
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_linux//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                       
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_linux_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                               
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_macos//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                       
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                               
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_win//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                         
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_win_arm64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                   
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_linux//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                       
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_linux_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                               
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_macos//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                       
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_win//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                         
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk18_win_arm64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                   
INFO: ToolchainResolution: Target platform @local_config_platform//:host: Selected execution platform @local_config_platform//:host, type @bazel_tools//tools/cpp:toolchain_type -> toolchain //tools/config:sophos_win_x64_toolchain, type @bazel_tools//tools/jdk:toolchain_type -> toolchain @bazel_tools//tools/jdk:toolchain_java8, type @bazel_tools//tools/jdk:runtime_toolchain_type -> toolchain @remotejdk11_win//:jdk          
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @local_jdk//:jdk; mismatching config settings: localjdk_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_linux//:jdk; mismatching values: linux, x86_64  
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_linux_aarch64//:jdk; mismatching values: linux, arm64                                                                                                                                                                                                                
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_linux_ppc64le//:jdk; mismatching values: linux, ppc                                                                                                                                                                                                                  
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_linux_s390x//:jdk; mismatching values: linux, s390x                                                                                                                                                                                                                  
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_macos//:jdk; mismatching values: osx, x86_64    
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_macos_aarch64//:jdk; mismatching values: osx, arm64                                                                                                                                                                                                                  
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_win//:jdk; mismatching values: x86_64           
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk11_win_arm64//:jdk; mismatching values: arm64      
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk17_linux//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                     
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk17_linux_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                             
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk17_macos//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk17_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                             
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk17_win//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                       
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk17_win_arm64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                 
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk18_linux//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                     
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk18_linux_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                             
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk18_macos//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                     
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk18_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                             
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk18_win//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                       
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: Rejected toolchain @remotejdk18_win_arm64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                                                 
INFO: ToolchainResolution:   Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform //tools/config:win_x86_platform: No toolchains found.   
```
```

**Comment by carpenterjc** [comment]
This does not reproduce in bazel 5.3.2, or 5.4.0rc1

This is reproducible in bazel 6.0.0.rc1

**Comment by comius** [comment]
It doesn't seem we ever had support for 32 bit windows:
https://github.com/bazelbuild/bazel/blob/5.3.2/src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE.tmpl

@carpenterjc
Can you provide what cpu and os constraints are you using in //tools/config:win_x86_platform. Are the additional constraints?
Can you provide output of toolchain resolution debug with bazel 5.3.2?

You should be able to register additional JDKs in your WORKSPACE with `remote_java_repository` following examples in jdk.WORKSPACE.tmpl.

**Comment by carpenterjc** [comment]
The platform is:

```platform(
    name = "win_x86_platform",
    constraint_values = [
        "@platforms//cpu:x86_32",
        "@platforms//os:windows",
        "@bazel_tools//tools/cpp:msvc",
    ],
)```

Toolchain resolution debug:
```
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @local_jdk//:jdk; mismatching config settings: localjdk_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: execution platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux//:jdk; mismatching values: linux
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: execution platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_aarch64//:jdk; mismatching values: linux, arm64
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: execution platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_ppc64le//:jdk; mismatching values: linux, ppc
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: execution platform @local_config_platform//:host: Rejected toolchain @remotejdk11_linux_s390x//:jdk; mismatching values: linux, s390x
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: execution platform @local_config_platform//:host: Rejected toolchain @remotejdk11_macos//:jdk; mismatching values: osx
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: execution platform @local_config_platform//:host: Rejected toolchain @remotejdk11_macos_aarch64//:jdk; mismatching values: osx, arm64
INFO: ToolchainResolution:   Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: execution @local_config_platform//:host: Selected toolchain @remotejdk11_win//:jdk
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk15_linux//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk15_macos//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk15_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk15_win//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk16_linux//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk16_macos//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk16_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk16_win//:jdk; mismatching config settings: prefix_version_setting
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_linux//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                   
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_macos//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                   
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_macos_aarch64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                           
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_win//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                                     
INFO: ToolchainResolution:     Type @bazel_tools//tools/jdk:runtime_toolchain_type: target platform @local_config_platform//:host: Rejected toolchain @remotejdk17_win_arm64//:jdk; mismatching config settings: prefix_version_setting                                                                                                                                                               
INFO: ToolchainResolution: Target platform @local_config_platform//:host: Selected execution platform @local_config_platform//:host, type @bazel_tools//tools/jdk:runtime_toolchain_type -> toolchain @remotejdk11_win//:jdk         
```

**Comment by carpenterjc** [comment]
cc_library - no failure
cc_test - Fails with toolchain resolution
cc_binary - no failure

**Comment by comius** [comment]
This was most likely introduced by changing exec_constraints to target_constraints in Java tools. I believe behavior of 6.0.0 is more correct.

You can resolve the problem by adding default java toolchain for windows x86 to your WORKSPACE. 

```
remote_java_repository(
    name = "remotejdk11_win_i686",
    exec_compatible_with = [
        "@platforms//os:windows",
        "@platforms//cpu:x86_32",
    ],
    sha256 = "421e03b0b70a515cb8e5994a6ceb86e7c5903c98383965c12a394d579ffc198b",
    strip_prefix = "zulu11.60.19-ca-jdk11.0.17-win_i686",
    urls = [
        "https://cdn.azul.com/zulu/bin/zulu11.60.19-ca-jdk11.0.17-win_i686.zip",
    ],
    version = "11",
)
```

I don't think we should add this by default to Bazel.

It's quite possible that the dependecies originate from coverage, if you don't have any other Java deps.

**Comment by carpenterjc** [comment]
I can confirm we do not use Java, and we never use coverage, as there isn't support form MSVC and windows. 

I can see this in the dependencies for a cc_test:
```
@remote_coverage_tools//:Main
@remote_coverage_tools//:all_lcov_merger_lib
@remote_coverage_tools//:all_lcov_merger_tools_deploy.jar
@remote_coverage_tools//:coverage_report_generator
@remote_coverage_tools//:lcov_merger
```

In this case I think there may be something wrong with the `@remove_coverage_tools`, because I would expect these tools to be built for the host system `cfg = "exec"` which in this case is intel x64. 

I am reluctant to add a dependencies on `rules_java` and an extra `jdk` which are not used.  It also looks like we now have to explicitly import `rules_java` because the built in `load("@bazel_tools//tools/jdk:remote_java_repository.bzl", "remote_java_repository")` complains that there is no `exec_compatable_with`.

**Comment by fmeum** [comment]
@comius I think I know what's wrong here, will submit a PR soon.

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by carpenterjc** [comment]
I found this issue, which may be adjacent. https://github.com/bazelbuild/bazel/issues/15088

**Comment by meteorcloudy** [comment]
@bazel-io fork 6.0.0

**Comment by comius** [comment]
If this was introduced by cc_test Starlarkification, this is a regression we should fix. I don't consider it critical for first Bazel LTS release as it's possible to workaround.

**Comment by fmeum** [comment]
@comius I submitted a fix for the 6.0.0 regression (https://github.com/bazelbuild/bazel/pull/16994). While I am at it, I will create a follow-up PR to fix https://github.com/bazelbuild/bazel/issues/15088, but that is of course not a regression and the fix should not go into 6.0.0.
